core/src/test/scala/unit/kafka/log/CleanerTest.scala

```diff
@@ -108,6 +108,37 @@ class CleanerTest extends JUnitSuite {
   }
 
   @Test
+  def testPartialSegmentClean() {
+    var cleaner = makeCleaner(2)
```

Comment:
Perhaps add a comment that since the loading factor is 0.75, the capacity is only 1 slot.